homeassistant/components/ccm15/climate.py

```diff
@@ -0,0 +1,166 @@
+            _LOGGER.debug("hvac_mode[%s]=%s", self._ac_index, mode)
+            return CONST_CMD_STATE_MAP[mode]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return fan mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            mode = data.fan_mode
+            _LOGGER.debug("fan_mode[%s]=%s", self._ac_index, mode)
+            return CONST_CMD_FAN_MAP[mode]
+        return None
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return swing mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("is_swing_on[%s]=%s", self._ac_index, data.is_swing_on)
+            return SWING_ON if data.is_swing_on else SWING_OFF
+        return None
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, f"{self._ac_host}.{self._ac_index}"),
+            },
+            name=self._attr_name,
+            manufacturer="Midea",
+            model="CCM15",
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the optional state attributes."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            return {"error_code": data.error_code}
+        return {}
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        await self.coordinator.async_set_temperature(self._ac_index, temperature)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set the hvac mode."""
+        await self.coordinator.async_set_hvac_mode(self._ac_index, hvac_mode)
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set the fan mode."""
+        await self.coordinator.async_set_fan_mode(self._ac_index, fan_mode)
+
+    async def async_turn_off(self) -> None:
+        """Turn off."""
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    async def async_turn_on(self) -> None:
+        """Turn on."""
+        await self.async_set_hvac_mode(HVACMode.AUTO)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all climate."""
+    coordinator: CCM15Coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+    ac_index = 0
+    ac_data: CCM15DeviceState = coordinator.data
+    for ac_index in ac_data.devices:
+        _LOGGER.debug("Creating new ac device at index '%s'", ac_index)
+        ac_device = CCM15Climate(coordinator.get_host(), ac_index, coordinator)
+        entities.append(ac_device)
+    async_add_entities(entities, True)
```

Comment:
Your coordinator already has data, update before add is not needed


```suggestion
    async_add_entities(entities)
```